ci/infra/testrunner/tests/test_node_reboot.py

```diff
@@ -4,58 +4,44 @@
 
 import pytest
 
-from tests.utils import wait
+from tests.utils import (check_pods_ready, node_is_ready, wait)
 
 from timeout_decorator import timeout
 
 logger = logging.getLogger("testrunner")
 
 
-def check_pods_ready(kubectl):
-    pods = json.loads(kubectl.run_kubectl('get pods --all-namespaces -o json'))['items']
-    for pod in pods:
-        pod_status = pod['status']['phase']
-        pod_name   = pod['metadata']['name']
-        assert pod_status in ['Running', 'Completed'], f'Pod {pod_name} status {pod_status} != Running or Completed'
+def check_node_is_ready(platform, kubectl, role, nr):
+    assert node_is_ready(platform, kubectl, role, nr)
 
 
 @pytest.mark.pr
-@pytest.mark.parametrize('node_type,node_number', [('master', 1), ('worker', 0)])
-def test_hard_reboot(deployment, platform, skuba, kubectl, node_type, node_number):
-    assert skuba.num_of_nodes(node_type) > node_number
+@pytest.mark.parametrize('role,node', [('master', 1), ('worker', 0)])
+def test_hard_reboot(deployment, platform, skuba, kubectl, role, node):
+    """ Reboots master and worker nodes and checks the are back ready.
+    For masters, don't master 1, as master 0 is expected to be the
+    cluster leader and rebooting it may introduce transient etcd errors.
+    """
+    assert skuba.num_of_nodes(role) > node
 
-    logger.info('Wait for all the nodes to be ready')
-    wait(kubectl.run_kubectl, 
-         'wait --for=condition=Ready node --all --timeout=0',
-         wait_timeout=10,
-         wait_backoff=30,
-         wait_elapsed=300,
-         wait_allow=(RuntimeError))
+    platform.ssh_run(role, node, 'sudo reboot &')
 
-    logger.info(f'Rebooting {node_type} {node_number}')
-
-    platform.ssh_run(node_type, node_number, 'sudo reboot &')
-
-    # wait the node to reboot
-    wait(platform.ssh_run,
-         node_type,
-         node_number,
-         'echo hello',
-         wait_backoff=30,
-         wait_timeout=30,
-         wait_elapsed=300,
-         wait_allow=(RuntimeError))
+    # Allow time for the node to reboot and kubernetes to check readiness
+    time.sleep(60)
 
     # wait the node to become ready
-    wait(kubectl.run_kubectl,
-         'wait --for=condition=Ready node --all --timeout=0',
+    wait(check_node_is_ready,
```

Comment:
couldn't this call fail if the network is not ready, assume this is only eating AssertionError exceptions. Do we get this sort of exception if the node is not reachable?

Reply:
No, it won't fail as we are calling the API server which is in HA setup.